docs/dbg.md

```diff
@@ -0,0 +1,162 @@
+# DBG helper
+
+
+### store
+
+`dbg.store` is a convenient way for getting the redux store.
+
+```js
+const store = dbg.store.getState()
+
+// select a source
+const source = dbg.store.getState().sources.sources.first();
+```
+
+### actions
+
+`dbg.actions` is a convenient way for getting the redux actions.
+The actions are bound so you can call them directly!
+
+```js
+const source = dbg.store.getState().sources.sources.first();
+dbg.actions.selectSource(source.get("id")))
+```
+
+### selectors
+
+`dbg.selectors` is a convenient way for getting the redux selectors.
+The selectors are bound so you can call them without `store.getState()`!
+
+```js
+const source = dbg.selectors.getSelectedSource();
+```
+
+### client
+
+`dbg.client` is a convenient way for getting the firefox commands.
+The commands are the interface for talking to the debugger server.
+
+```js
+const source = dbg.selectors.getSelectedSource();
+dbg.client
+  .setBreakpoint({line: 24, sourceId: source.get("id")})
+  .then(console.log)
+```
+
+### connection
+
+`dbg.connection` has several useful properties for seeing the current connection to the tab.
+If you're looking for the current threadClient, debuggerClient, or tabTarget it is your friend!
+
+```js
+dbg.connection.tab
+/*
+{
+  "title": "Backbone.js • TodoMVC",
+  "url": "http://firefox-dev.tools/debugger-examples/examples/todomvc/",
+  "id": "server1.conn12.child1/tab1",
+  "tab": {
+    "actor": "server1.conn12.child1/tab1",
+    "title": "Backbone.js • TodoMVC",
+    "url": "http://firefox-dev.tools/debugger-examples/examples/todomvc/",
+    "outerWindowID": 2147483649,
+    "consoleActor": "server1.conn12.child1/consoleActor2",
+    "inspectorActor": "server1.conn12.child1/inspectorActor3",
+    "callWatcherActor": "server1.conn12.child1/callWatcherActor4",
+  },
+  "clientType": "firefox"
+}
+*/
+
+dbg.connection.connTarget // { type: "firefox", param: "child1/tab1" }
+
+dbg.connection.tabConnection // debuggerClient, threadClient, tabTarget
+```
+
+### bpClients
+
+`dbg.bpClients` stores the current set of breakpoint clients.
+
+### services
+
+`dbg.services` has a reference to the SourceMap worker, which is available to the redux actions.
+
+`dbg.services.sourceMaps`
+
+> originalToGeneratedId, generatedToOriginalId, isGeneratedId, isOriginalId, hasMappedSource, getOriginalURLs, getGeneratedLocation, getOriginalLocation, getLocationScopes, getOriginalSourceText, applySourceMap, clearSourceMaps, startSourceMapWorker, stopSourceMapWorker
+
+### prefs
+
+`dbg.prefs` references the PreferencesHelper. You can use `dbg.prefs` to see or change the state of any pref.
+
+```js
+dbg.prefs.pauseOnExceptions // false
+dbg.prefs.pauseOnExceptions = true
+dbg.prefs.pauseOnExceptions // true
+```
+
+### features
+
+`dbg.features` references the debugger's feature flags. You can use `dbg.features` to see or change the state of any flag.
```

Comment:
"`dbg.features` references the debugger's feature flags." is a great descriptive sentence about what features is!